### ISSUE TITLE
[stable/prometheus-operator] support podmonitors

### ISSUE THREAD

**Issue by JoshuaC215** [helm/charts#16119]
**Is your feature request related to a problem? Please describe.**
Prometheus Operator recently [added support for PodMonitors](https://github.com/coreos/prometheus-operator/blob/0ddafadf0376168a46023316ace5fe1e502cefc8/Documentation/api.md#podmonitor), but currently no pod monitor selector can be configured via the chart, making it difficult (or impossible?) to use with a helm install of prometheus-operator. Pod Monitors seem super useful! So it would be great to support them.

**Describe the solution you'd like**
I believe we can add a straightforward configuration for `podMonitorSelector` / `podMonitorNamespaceSelector` in the [prometheus](https://github.com/coreos/prometheus-operator/blob/0ddafadf0376168a46023316ace5fe1e502cefc8/Documentation/api.md#prometheusspec) template which will enable most basic support while this feature is still experimental. Long term might want to add something comparable to `prometheus.additionalServiceMonitors` but I don't think it's immediately necessary to unblock use of this feature.

**Describe alternatives you've considered**
Not sure of any readily available workaround other than manually updating prometheus config outside of helm.

**Additional context**
N/A. I don't mind taking a crack at this if maintainers are supportive of a very simple design.

**Comment by JoshKCarroll** [comment]
(Also if there's any suggested workaround that I'm missing let me know!)

**Comment by andresua** [comment]
What I did in the meantime was that installed the chart (`helm install`) and I added (via `kubectl get/delete/create`):
```yaml
podMonitorNamespaceSelector: {}
  podMonitorSelector:
    matchLabels:
      release: prometheus-operator
```
After the *ServiceMonitor* section in the **Prometheus** *prometheus-operator-prometheus* (also via `kubectl`).
Then I created my **PodMonitor** (again via `kubectl`) and **Grafana** displayed my custom metrics right away.

Just to check that it was correctly loaded, I went to see the configuration in the console of **Prometheus** and it loaded my scraping definition with `role: pod`.

Also, in the repo of Prometheus, they say that *PodMonitor* is experimental which may be the reason why helm hasn't included it yet.

**Comment by bashofmann** [comment]
There is already a pull request for this: https://github.com/helm/charts/pull/15928

**Comment by JoshKCarroll** [comment]
I see this was merged in today with version 6.12.0. I'm planning to test it out in our environment this week and then will close this unless someone else gets to it first :) thanks!!!

**Comment by bashofmann** [comment]
Tested it and PodMonitors work as expected now.
